src/Utilities/TypeTraits.hpp

```diff
@@ -1296,6 +1296,19 @@ template <typename T>
 struct remove_reference_wrapper<std::reference_wrapper<T>> {
   using type = T;
 };
+template <typename T>
```

Comment:
Please remove this, this is not how it was intended to be used.

Reply:
As far as I see everything in the STL type_traits header ignores top level cv qualifiers (except for the things that explicitly reference them).  Do you have an intended use case for the atypical behavior?

Reply:
I would argue that the use case Gabe proposes is atypical...

Reply:
If this is a desired change I then every use case of reference_wrapper should be checked for const correctness, though that might be necessary anyway.

Reply:
I'm not asking whether use cases are typical or not.  The actual behavior of the type trait doesn't match the standard type trait behavior in the STL, which makes the type trait itself atypical.  I think we should match the behavior of similar stuff in the STL unless we have a good reason to do otherwise.

Reply:
Well, since const reference_wrappers return non-const references it seems reference_wrapper is the weird thing here. The use cases in the code seem to be correct, so sure the change is fine, but not without tests.